components/ai_chat/resources/page/components/error_rate_limit/index.tsx

```diff
@@ -8,26 +8,72 @@ import { getLocale } from '$web-common/locale'
 import Icon from '@brave/leo/react/icon'
 import Button from '@brave/leo/react/button'
 
+import DataContext from '../../state/context'
 import styles from './style.module.scss'
 
 interface ErrorRateLimit {
   onRetry?: () => void
 }
 
-function ErrorRateLimit (props: ErrorRateLimit) {
+function ErrorRateLimitPremium(props: ErrorRateLimit) {
+  const pricing = getLocale('premiumPricing').split('/')
+
   return (
-    <div className={styles.box}>
-      <Icon name="warning-triangle-filled" className={styles.icon} />
-      <div>
-        <p>{getLocale('errorRateLimit')}</p>
-        <div className={styles.actionsBox}>
-          <Button onClick={props.onRetry}>
-            {getLocale('retryButtonLabel')}
-          </Button>
-        </div>
+    <div className={styles.boxPremium}>
+      <h4>{getLocale('rateLimitReachedTitle')}</h4>
+      <p>{getLocale('rateLimitReachedDesc')}</p>
+      <ul className={styles.featuresListing}>
+        <li>
+          <Icon name='check-normal' />
+          <span>{getLocale('premiumFeature_1')}</span>
+        </li>
+        <li>
+          <Icon name='check-normal' />
+          <span>{getLocale('premiumFeature_2')}</span>
+        </li>
+        <li className={styles.priceList}>
+          <span>{getLocale('premiumLabel')}</span>
+          <span className={styles.price}>
+            {pricing[0]}
+            <data>{pricing[1]}</data>
+            {pricing[2]}
+          </span>
+        </li>
+      </ul>
+      <div className={styles.actionsBox}>
+        <Button onClick={props.onRetry}>
+          {getLocale('premiumButtonLabel')}
+        </Button>
+        <Button kind='plain-faint' onClick={props.onRetry}>
+          {getLocale('retryButtonLabel')}
+        </Button>
       </div>
```

Comment:
componentize

Reply:
I have componentized it and only exposed the props that's needed for this PR.